### PR TITLE
PIM-9843 pull_up_40 to 50

### PR DESCRIPTION
**Pull request by rfauglas**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/EnsureLockTableExistsQuery.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Bundle\Doctrine\ORM\Query;
+
+use Akeneo\Pim\Enrichment\Component\Lock\Query\EnsureLockTableExistsInterface;
+use Doctrine\DBAL\Connection;
+
+/**
+ * TODO pull up remove this
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/EnsureLockTableExistsQuery.php:11]: Does this concern only master or also 5.0? cc @willy-ahva

**Review by jmleroux — APPROVED** [review]
GTM after check

File: src/Akeneo/Pim/Enrichment/Bundle/Resources/config/queries.yml [hunk 2]
```diff
@@ -312,6 +312,11 @@ services:
         arguments:
             - '@database_connection'
 
+    akeneo.pim.enrichment.query.ensure_lock_table_exists:
+        class: Akeneo\Pim\Enrichment\Bundle\Doctrine\ORM\Query\EnsureLockTableExistsQuery
```
Comment (jmleroux) [inline src/Akeneo/Pim/Enrichment/Bundle/Resources/config/queries.yml:316]: A new service without the new class in the PR. Smells like this service should not be there.

**Review by jmleroux — APPROVED** [review]
CI will tell

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/EnsureLockTableExistsQuery.php (hunk 1 above)
Comment (rfauglas) [reply src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/EnsureLockTableExistsQuery.php]: Removed!

**Pull request closed without merging**